### ISSUE TITLE
SONiC 202205 build fails to retrieve hiredis_0.14.0-3~bpo9+1.dsc

### ISSUE THREAD

**Issue by spilkey-cisco** [sonic-net/sonic-buildimage#15588]
#### Description

hiredis_0.14.0-3~bpo9+1.dsc does not exist in http://cdn-fastly.deb.debian.org/debian/pool/main/h/hiredis
Looks like it was moved from main to archive: http://archive.debian.org/debian/pool/main/h/hiredis/

```
Location: http://cdn-fastly.deb.debian.org/debian/pool/main/h/hiredis/hiredis_0.14.0-3~bpo9+1.dsc [following]
--2023-06-20 21:48:26--  http://cdn-fastly.deb.debian.org/debian/pool/main/h/hiredis/hiredis_0.14.0-3~bpo9+1.dsc
2023-06-20 21:48:26 ERROR 404: Not Found.
```

#### Steps to reproduce the issue:
`make all` of sonic-buildimage 202205 branch

Removing `~bpo9+1` from https://github.com/sonic-net/sonic-buildimage/blob/202205/rules/hiredis.mk#L4 was used as a workaround, and the build succeeded.

Similar to issue https://github.com/sonic-net/sonic-buildimage/issues/14854

**Comment by chidanandg** [comment]
Yes, this link http://archive.debian.org/debian/pool/main/h/hiredis/ worked for me. 
I modified the hiredis Makefile with the said link and my sonic build went through. 

I am using older version of sonic (from  Jun 2022).   Would be right to change the Makefile? Is there any other generic option ?
